depfinder/main.py

```diff
@@ -70,11 +70,24 @@
     Loader=yaml.SafeLoader,
 )
 
+with open('depfinder/pkg_data/name_mapping.yaml', 'r') as f:
+   mapping_list = yaml.load(f, Loader= yaml.SafeLoader)
 
-def _split(name):
-    named_space = pkg_data['_NAMEDSPACE_MAPPING'].get(name)
-    return named_space if named_space else name.split('.')[0]
+namespace_mapping = {}
+for pkg in mapping_list:
+   if pkg['import_name'] != pkg['conda_name']:
+       namespace_mapping[pkg['import_name']] = pkg['conda_name']
 
+def _split(name):
+    mapped = namespace_mapping.get(name)
+    if mapped:
+        return mapped
+    else:
+        if not '.' in name:
+            return name
+        else:
+            pieces = name.split('.')[:-1]
```

Comment:
I think you might want to use `rsplit` here.

Reply:
but isn't split easier with join?

Reply:
rsplit doesn't need to be joined, since you can request it to only split off the right most `.`